api/src/main/java/org/pmiops/workbench/workspaces/resources/UserRecentResourceServiceImpl.java

```diff
@@ -81,18 +55,6 @@ public DbUserRecentlyModifiedResource updateNotebookEntry(
    */
   @Override
   public void updateCohortEntry(long workspaceId, long userId, long cohortId) {
-    Timestamp now = new Timestamp(clock.instant().toEpochMilli());
-
-    DbCohort cohort = cohortDao.findById(cohortId).orElse(null);
-    DbUserRecentResource resource =
-        userRecentResourceDao.findByUserIdAndWorkspaceIdAndCohort(userId, workspaceId, cohort);
-    if (resource == null) {
-      handleUserLimit(userId);
```

Comment:
we would need to include handleUserLimit(userId) for cohort and concept set.

Reply:
Yes, good catch.  Moved the check to the common method for all resource types, `updateUserRecentlyModifiedResourceEntry()`